pkg/error/error.go

```diff
@@ -20,6 +22,13 @@ type ValidationErrors struct {
 	Errors   []ValidationError
 }
 
+func (error1 *ValidationError) Equal(error2 ValidationError) bool {
+	return error1.Message.Error() == error2.Message.Error() &&
+		error1.LineNumber == error2.LineNumber &&
+		error1.AdditionalIdenticalErrorCount == error2.AdditionalIdenticalErrorCount
+
+}
```

Comment:
This code might be refactored (it's up to you)

To something like this,making it more readable 


```suggestion
func (error1 *ValidationError) Equal(error2 ValidationError) bool {
	if error1.LineNumber != error2.LineNumber {
		return false
	}

	if error1.Message.Error() != error2.Message.Error() {
		return false
	}

	return true
}
```

I inverted the test on message and line, comparing integers is faster

I removed the test on count, because I don't think you [need](https://github.com/editorconfig-checker/editorconfig-checker/pull/360#discussion_r1739717042) it

Reply:
I didn't know that early exits are idiomatic golang - but I definitely like them to make code more readable. I'll implement that!

Reply:
rewritten in e16817eefdda4c9a425931d2b47f53f030a36fed

Reply:
Idiomatic or not the readability always should come first.

A long line with 3 conditions is never a good thing 😅😂